File: frontend/src/types/network.ts
export interface Connection {
  local: string
  remote: string
  process: string
  internal: boolean
}

export interface HistoryPoint {
  time: string
  outbound: number
}

export interface NetworkStatus {
  outbound: number
  airGapped: boolean
  connections: Connection[]
  history: HistoryPoint[]
}

File: backend/app/schemas/network.py
from app.schemas.base import CamelModel


class ConnectionOut(CamelModel):
    local: str
    remote: str
    process: str
    internal: bool


class HistoryPoint(CamelModel):
    time: str
    outbound: int


class NetworkStatus(CamelModel):
    outbound: int
    air_gapped: bool
    connections: list[ConnectionOut]
    history: list[HistoryPoint]